src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/scarlet_monastery/boss_mograine_and_whitemane.cpp

```diff
@@ -126,8 +128,11 @@ struct boss_scarlet_commander_mograineAI : public ScriptedAI
             return;
 
         Creature* pWhitemane = m_pInstance->GetSingleCreatureFromStorage(NPC_WHITEMANE);
-        if (pWhitemane && !pWhitemane->IsAlive())
+        if (pWhitemane && !pWhitemane->IsAlive()){
+            if (!(m_pInstance->GetData(TYPE_MOGRAINE_AND_WHITE_EVENT) == NOT_STARTED) || !(m_pInstance->GetData(TYPE_MOGRAINE_AND_WHITE_EVENT) == FAIL))
+                m_pInstance->SetData(TYPE_MOGRAINE_AND_WHITE_EVENT, FAIL);
             pWhitemane->Respawn();
+        }            
     }
 
     void DamageTaken(Unit* /*dealer*/, uint32& damage, DamageEffectType /*damagetype*/, SpellEntry const* /*spellInfo*/) override
```

Comment:
I suggest removing this func per rework and use the new SetDeathPrevention paradigm and new hook JustPreventedDeath.

Reply:
Done